### PR TITLE
Add scala.js support

### PR DESCRIPTION
**Pull request by InTheNow**
Also uses scalatest 3.0.0-M7

### CODE REVIEW

File: shared/src/main/scala/Laws.scala [hunk 1]
```diff
@@ -80,15 +80,7 @@ trait Laws {
...
-    final def all: Properties = new Properties(name) {
-      for {
-        (baseName, baseProps) ← bases.sortBy(_._1)
-        (name, prop) ← baseProps.all.properties
-      } property(baseName + ":" + name) = prop
-
-      for ((name, prop) ← collectParentProps)
-        property(name) = prop
-    }
+   final def all: Properties = new AllProperties(name , bases ,collectParentProps)
```
Comment (larsrh) [inline shared/src/main/scala/Laws.scala]: What's the reason for this change?

File: build.sbt [hunk 2]
```diff
@@ -8,30 +8,99 @@ import com.typesafe.sbt.pgp.PgpKeys._
...
-
-libraryDependencies ++= Seq(
-  "org.scalacheck" %% "scalacheck"        % "1.12.4",
-  "org.scalatest"  %% "scalatest"         % "2.2.5" % "optional",
-  "org.specs2"     %% "specs2-core"       % "3.6" % "optional",
-  "org.specs2"     %% "specs2-scalacheck" % "3.6" % "optional"
-)
+name := "discipline root project"
+
+crossScalaVersions := Seq("2.10.4", "2.11.6")
```
Comment (larsrh) [inline build.sbt]: 2.10.5?

File: shared/src/main/scala/Laws.scala (hunk 1 above)
Comment (InTheNow) [inline shared/src/main/scala/Laws.scala]: Scala.js does not support nested classes - see https://github.com/scala-js/scala-js/issues/354

File: build.sbt (hunk 2 above)
Comment (InTheNow) [inline build.sbt]: OK - if all else is OK I will change, squash and push

Comment (larsrh) [inline build.sbt]: Yes, looks fine! Thanks for your effort.

**Comment by InTheNow** [comment]
Also changed to 2.11.7

**Comment by InTheNow** [comment]
Oh - and thanks for doing this as well :)

**Comment by larsrh** [comment]
Is there any way to check whether the tests are passing using a Javascript runtime?

**Comment by InTheNow** [comment]
The tests are checked in JS - scalatest is now x-compiling. Use `disciplineJS/test` to run just the JS tests
